src/include/data/data_batch_view.hpp

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2025, Sirius Contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+#include <variant>
+#include <memory>
+#include <cudf/table/table.hpp>
+
+#include "helper/helper.hpp"
+#include "data/common.hpp"
+
+namespace sirius {
+
+using sirius::memory::Tier;
+
+class DataBatchView : public cudf::table_view {
+public:
+    DataBatchView(DataBatch* batch, std::vector<cudf::column_view> const& cols)
```

Comment:
I think the constructor looks more like DataBatchView(DataBatch* batch)

We should do something which is as follows

{
 acquire some kind of lock
ensure the data is currently in gpu if its not we did something wrong
increment the reference count
releasee the lock
}
make the view from the data in the databatch. now come to think of it. Our databatch could have a method which creates a DataBatchView and it can do the updating of the reference count when it creates this view. But we want the movement of this data to the gpu to have happened previously and asynchronously, if this data batch is not in gpu it should be an error

}

Reply:
Since it's a derived class from cudf::table_view, it has to pass the constructor to the cudf::table_view right? Or are you thinking about something that is not a derived class of cudf::table_view?

Reply:
so do something like batch->CreateView()

Reply:
Ok I see where some of the confusion is coming from. So my main point is that cols comes from the data that resides inside of batch,so if batch has a cudf_table call the table->view() method and thats your view